private/protocol/query/unmarshal_error.go

```diff
@@ -36,5 +45,22 @@ func UnmarshalError(r *request.Request) {
 			r.HTTPResponse.StatusCode,
 			reqID,
 		)
+		return
+	}
+
+	// Check for unhandled error
+	servUnavailResp := xmlServiceUnavailableResponse{}
+	unavailErr := xml.Unmarshal(bodyBytes, &servUnavailResp)
```

Comment:
Wouldn't it be possible to just update
```go
type xmlErrorResponse struct {
	XMLName   xml.Name `xml:"ErrorResponse,ServiceUnavailableException"`
	Code      string   `xml:"Error>Code"`
	Message   string   `xml:"Error>Message"`
	RequestID string   `xml:"RequestId"`
}
```
then check the name rather than unmarshal twice?